$OpenBSD: patch-src_suricata_c,v 1.1 2019/03/05 12:38:24 bluhm Exp $

Fix pid file removal if it is specified in config file.
Use setresuid/gid() directly to change user and group.  Otherwise
Suricata uses libcap-ng on Linux and runs as root elsewhere.

Index: src/suricata.c
--- src/suricata.c.orig
+++ src/suricata.c
@@ -399,6 +399,8 @@ static void GlobalsDestroy(SCInstance *suri)
     MpmHSGlobalCleanup();
 #endif
 
+    /* The pid file name may be in config memory, but is needed later. */
+    suri->pid_filename = strdup(suri->pid_filename);
     ConfDeInit();
 #ifdef HAVE_LUAJIT
     LuajitFreeStatesPool();
@@ -408,6 +410,8 @@ static void GlobalsDestroy(SCInstance *suri)
     SCThresholdConfGlobalFree();
 
     SCPidfileRemove(suri->pid_filename);
+    free((char *)suri->pid_filename);
+    suri->pid_filename = NULL;
 }
 
 /** \brief make sure threads can stop the engine by calling this
@@ -3023,6 +3027,7 @@ int main(int argc, char **argv)
 #endif
 #endif
 
+    SCSetUserID(suricata.userid, suricata.groupid);
     SuricataMainLoop(&suricata);
 
     /* Update the engine stage/status flag */
